addons/l10n_gr_edi/models/account_move.py

```diff
@@ -0,0 +1,611 @@
+            ))
+
+    ################################################################################
+    # Dynamic Selection Field Computes
+    ################################################################################
+
+    @api.depends('move_type')
+    def _compute_l10n_gr_edi_available_inv_type(self):
+        for move in self:
+            if move.move_type in ('out_invoice', 'out_refund'):
+                move.l10n_gr_edi_available_inv_type = ','.join(INVOICE_TYPES_HAVE_INCOME)
+            else:
+                move.l10n_gr_edi_available_inv_type = ','.join(INVOICE_TYPES_HAVE_EXPENSE)
+
+    @api.depends('fiscal_position_id', 'l10n_gr_edi_available_inv_type')
+    def _compute_l10n_gr_edi_inv_type(self):
+        for move in self:
+            if move._origin.l10n_gr_edi_inv_type:
+                # get the real current inv type, needed for calculating line's cls category & type
+                move.l10n_gr_edi_inv_type = move._origin.l10n_gr_edi_inv_type
+            elif move.move_type in ('out_refund', 'in_refund'):
+                # inv_type specific for credit notes
+                if move.l10n_gr_edi_mark:
+                    # when possible, we must add the associate invoice/bill mark (id)
+                    move.l10n_gr_edi_inv_type = '5.1'
+                else:
+                    move.l10n_gr_edi_inv_type = '5.2'
+            else:
+                inv_type = '1.1' if move.move_type == 'out_invoice' else '13.1'
+                preferred_cls_ids = move.fiscal_position_id.l10n_gr_edi_preferred_classification_ids.filtered(
+                    lambda p: p.l10n_gr_edi_inv_type in move.l10n_gr_edi_available_inv_type.split(','))
+                if preferred_cls_ids:
+                    inv_type = preferred_cls_ids[0].l10n_gr_edi_inv_type
+                move.l10n_gr_edi_inv_type = inv_type
+
+    @api.depends('l10n_gr_edi_inv_type')
+    def _compute_l10n_gr_edi_need_fields(self):
+        for move in self:
+            move.l10n_gr_edi_need_correlated = move.l10n_gr_edi_inv_type in TYPES_WITH_CORRELATE_INVOICE
+            move.l10n_gr_edi_need_payment_method = move.l10n_gr_edi_inv_type in TYPES_WITH_MANDATORY_PAYMENT
+
+    @api.depends('l10n_gr_edi_need_correlated')
+    def _compute_l10n_gr_edi_correlation_id(self):
+        for move in self:
+            if move.l10n_gr_edi_need_correlated:
+                move.l10n_gr_edi_correlation_id = move.l10n_gr_edi_correlation_id
+
+                # automatically compute the correlated move id
+                if move.move_type in ('out_refund', 'in_refund') and (correlated_move := self.env['account.move'].search([
+                    ('company_id', '=', move.company_id.id),
+                    ('l10n_gr_edi_mark', '=', move.l10n_gr_edi_mark),
+                    ('move_type', '=', 'out_invoice' if move.move_type == 'out_refund' else 'in_invoice'),
+                ], limit=1)):
+                    move.l10n_gr_edi_correlation_id = correlated_move
```

Comment:
It's ugly imo :sweat: 
```suggestion
                if move.move_type not in ('out_refund', 'in_refund'):
                    continue
                    
                move.l10n_gr_edi_correlation_id = self.env['account.move'].search([
                    ('company_id', '=', move.company_id.id),
                    ('l10n_gr_edi_mark', '=', move.l10n_gr_edi_mark),
                    ('move_type', '=', 'out_invoice' if move.move_type == 'out_refund' else 'in_invoice'),
                ], limit=1))
```
You are doing a search in a loop. It would be nice to batch it somehow if possible